src/ConfigurationFactory.php

```diff
@@ -173,6 +173,15 @@ protected function createWithConfig(ContainerInterface $container, string $confi
             $configuration->setSQLLogger($config['sql_logger']);
         }
 
+        if ($config['middlewares'] !== []) {
+            $middlewares = [];
+            foreach ($config['middlewares'] as $middleware) {
+                $middlewares[] = is_string($middleware) ? $container->get($middleware) : $middleware;
```

Comment:
`$middleware` should be assumed to be a `string` here - an `assert()` is preferable to an `is_string()` check